### PR TITLE
[Console] Keep the original console as `global.originalConsole`

### PR DESCRIPTION
**Pull request by ide**
If a console exists, keep the original as `global.originalConsole` before overwriting `global.console` with a polyfill. This matches what we do for XHR, fetch, and some other libraries.

Test Plan: console.log still tees to both consoles. `originalConsole.log` prints only to the Web Inspector console in JSC.

### CODE REVIEW

**Comment by ide** [comment]
Use case for this is logging heavy data. JSC -> Web Inspector logging doesn't block the bridge as much.

**Comment by satya164** [comment]
cc @mkonicek

**Comment by mkonicek** [comment]
I have zero context on this, looks good though. Feel free to shipit.

**Comment by ide** [comment]
Thanks @mkonicek. We've been using this for several weeks without problems, so I don't expect this to break anything.

@facebook-github-bot shipit

File: packager/react-packager/src/Resolver/polyfills/console.js [hunk 1]
```diff
@@ -469,17 +474,19 @@
       trace: getNativeLogFunction(LOG_LEVELS.trace),
       table: consoleTablePolyfill
     };
+    descriptor.value = console;
```
Comment (davidaurelio) [inline packager/react-packager/src/Resolver/polyfills/console.js]: what if `!descriptor`?

**Comment by davidaurelio** [comment]
Is there a reason why we are using the whole descriptor magic instead of just assigning? I’m asking because this broke our internal tests on iOS 7.

**Comment by ide** [comment]
The descriptor was copied to preserve the property attributes like whether 'console' should be enumerable, for example. Just to keep the JS environment a little closer to its original state.

**Comment by davidaurelio** [comment]
I see. The actual problem was that on iOS 7, the `value` property of the descriptor is non-writable :-P

**Pull request closed without merging**